operator/pkg/util/merge_iop.go

```diff
@@ -275,6 +275,11 @@ func OverlayIOP(base, overlay string) (string, error) {
 
 	merged, err := strategicpatch.StrategicMergePatch(bj, oj, &iopMergeStruct)
 	if err != nil {
+		if strings.Contains(err.Error(), "unable to find api field in struct") {
+			i := strings.Index(err.Error(), "struct")
+			errMsg := err.Error()[i+7:]
+			return "", fmt.Errorf("%s is not supported in current version", errMsg)
```

Comment:
Is it possible to display the version number?
```suggestion
			return "", fmt.Errorf("%s is not supported from %s version", errMsg, version)
```
```
error: externalComponentSpec for the json field "grafana" is not supported from version 1.7.
```

Reply:
yes, it's possible, then i should propagate this error up to the upgrade() func to use the retrieved `targetVersion` there.